### PR TITLE
[ESL Alert] Introduce new ESL Alert component

### PR DESCRIPTION
**Pull request by julia-murashko**

### CODE REVIEW

File: src/modules/esl-alert/core/esl-alert.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+  public static init() {
+    if (document.querySelector(ESLAlert.is)) return;
+    ESLAlert.register();
+    const alert = document.createElement(ESLAlert.is) as ESLAlert;
+    document.body.appendChild(alert);
+  }
+
+  @bind
+  onTriggerHandler(e: CustomEvent) {
```
Comment (ala-n) [inline src/modules/esl-alert/core/esl-alert.ts:31]: Cosmetic: naming. onRequest / onControlEvent or smth like that

File: src/modules/esl-alert/core/esl-alert.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    if (e.type === `${ESLAlert.eventNs}:show`) {
+      const params = Object.assign({}, {hideDelay: 2500}, e.detail, {force: true});
```
Comment (ala-n) [inline src/modules/esl-alert/core/esl-alert.ts:33]: Magic number in the method is not good, mb we can introduce "Default" constant somewhere?

File: src/modules/esl-alert/core/esl-alert.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    super.unbindEvents();
+
+    window.removeEventListener(`${ESLAlert.eventNs}:show`, this.onTriggerHandler);
+    window.removeEventListener(`${ESLAlert.eventNs}:hide`, this.onTriggerHandler);
+  }
+}
+
+function createIframe() {
+  const iframe = document.createElement('iframe');
+  iframe.className = 'ie-zindex-fix';
```
Comment (ala-n) [inline src/modules/esl-alert/core/esl-alert.ts:80]: There is no such style in ESL right now

File: src/modules/esl-alert/core/esl-alert.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  cls?: string;
+}
+
+@ExportNs('Alert')
+export class ESLAlert extends ESLBasePopup {
+  static is = 'esl-alert';
+  static eventNs = 'esl:alert';
+
+  @jsonAttr<AlertActionParams>({defaultValue: {text: '', cls: ''}})
+  public defaultParams: AlertActionParams;
```
Comment (ala-n) [inline src/modules/esl-alert/core/esl-alert.ts:18]: Feature Request: There is no possibility for the final consumer to setup default without inheritance to introduce a modified component.
mb let's try to move defaultValue to the static object then.

File: src/modules/esl-alert/core/esl-alert.less [hunk 5]
```diff
@@ -0,0 +1,20 @@
+esl-alert {
```
Comment (ala-n) [inline src/modules/esl-alert/core/esl-alert.less:1]: Do this style super abstract?
We can not rely on consumer X here, We should define something really abstract for consuming.

File: src/modules/esl-alert/core/esl-alert.ts [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+      super.onShow(params);
+    }
+    this.hide(params);
+    return this;
+  }
+
+  protected connectedCallback() {
+    super.connectedCallback();
+    this.text = document.createElement('span');
+    this.text.className = 'hpe-alert-text';
```
Comment (ala-n) [inline src/modules/esl-alert/core/esl-alert.ts:55]: Really really bed class name!

File: src/modules/esl-alert/core/esl-alert.ts [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+@ExportNs('Alert')
+export class ESLAlert extends ESLBasePopup {
+  static is = 'esl-alert';
+  static eventNs = 'esl:alert';
+
+  @jsonAttr<AlertActionParams>({defaultValue: {text: '', cls: ''}})
+  public defaultParams: AlertActionParams;
+
+  @attr({defaultValue: ''}) public cls: string;
```
Comment (ala-n) [inline src/modules/esl-alert/core/esl-alert.ts:20]: ???
Class is a CSS class that will be added via CSSUtility but not a type attribute that we use in a prototype.

**Review by ala-n — CHANGES_REQUESTED** [review]

**Comment by ala-n** [comment]
Also full code documentation.
And separately or with this PR documentation README.

**Review by ala-n — APPROVED** [review]

**Review by yadamskaya — APPROVED** [review]

**Pull request merged**